### DISCUSSION TITLE
Custom setter with focusAtom

### DISCUSSION THREAD

**Discussion by KorySchneider** [pmndrs/jotai#1950] [Q&A] [answered]
Hello,

I am wondering if I can use a custom setter with `focusAtom`.

Consider this small, contrived use case:

```
const baseAtom = atom({
  foo: '',
  bar: true,
});

const fooAtom = focusAtom(baseAtom, o => o.prop('foo'));
const barAtom = focusAtom(baseAtom, o => o.prop('bar'));
```

How can I  implement a custom setter so that `bar` gets set to `false` whenever the value of `foo` changes?

Thanks!

**Comment by dai-shi** [comment, marked as answer]
Regardless of focusAtom, you can't change an atom value based on another atom value changes, because it's technically a side effect. You want to create a custom setter that updates both at the same time.

**Reply by KorySchneider** [reply]
Ah that makes sense, thank you.

To make sure I'm understanding, and to help those who might come across this, you mean a helper function like `updateFoo` in this example below?

```
const setFoo = useSetAtom(fooAtom);
const setBar = useSetAtom(barAtom);

const updateFoo = (newFoo) => {
  setFoo(newFoo);
  setBar(false);
}
```

**Reply by dai-shi** [reply]
Exactly.

In this specific example, you could directly update `baseAtom`too.
You could also create write-only atom instead of using hooks.
